lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+   * @method getTranslation
+   * @return {p5.Vector}
+   */
+  p5.Transform2D.prototype.getTranslation = function() {
+    return new p5.Vector(this[2], this[5]);
+  };
+
+  /**
+   * Modify this transform, scaling it by a certain amount.
+   * Returns this transform.
+   * @method scale
+   * @return {p5.Transform2D}
+   * @example
+   *     // Three different ways to call this method.
+   *     var t = new p5.Transform();
+   *     // 1. One scalar value
+   *     t.scale(uniformScale);
+   *     // 1. Two scalar values
+   *     t.scale(scaleX, scaleY);
+   *     // 2. One vector
+   *     t.translate(new p5.Vector(scaleX, scaleY));
+   */
+  p5.Transform2D.prototype.scale = function(arg0, arg1) {
+    var sx, sy;
+    if (arg0 instanceof p5.Vector) {
+      sx = arg0.x;
+      sy = arg0.y;
+    } else if (typeof arg0 === 'number' && typeof arg1 === 'number') {
+      sx = arg0;
+      sy = arg1;
+    } else if (typeof arg0 === 'number') {
+      sx = arg0;
+      sy = arg0;
+    } else {
+      throw new TypeError('Invalid arguments to Transform2D.scale: ' + arguments);
+    }
+    return p5.Transform2D.mult([
+      sx, 0, 0,
+      0, sy, 0
+    ], this, this);
+  };
+
+  /**
+   * Retrieve the scale vector of this transform.
+   * @method getScale
+   * @return {p5.Vector}
+   */
+  p5.Transform2D.prototype.getScale = function() {
+    var a = this[0], b = this[1],
+        c = this[3], d = this[4];
+    return new p5.Vector(
+      sign(a) * Math.sqrt(a*a + b*b),
+      sign(d) * Math.sqrt(c*c + d*d)
+    );
+  };
+
+  /*
+   * Return -1, 0, or 1 depending on whether a number is negative, zero, or positive.
+   */
+  function sign(x) {
+    x = +x; // convert to a number
+    if (x === 0 || isNaN(x)) {
+      return Number(x);
+    }
+    return x > 0 ? 1 : -1;
+  }
+
+  /**
+   * Modify this transform, rotating it by a certain amount.
+   * @method rotate
+   * @param {number} radians
+   * @return {p5.Transform2D}
+   */
+  p5.Transform2D.prototype.rotate = function(radians) {
+    // Clockwise!
+    if (typeof radians !== 'number') {
+      throw new TypeError('Invalid arguments to Transform2D.rotate: ' + arguments);
+    }
+    var sinR = Math.sin(radians);
+    var cosR = Math.cos(radians);
+    return p5.Transform2D.mult([
+      cosR, -sinR, 0,
+      sinR, cosR, 0
+    ], this, this);
+  };
+
+  /**
+   * Retrieve the angle of this transform in radians.
+   * @method getRotation
+   * @return {number}
+   */
+  p5.Transform2D.prototype.getRotation = function() {
+    // see http://math.stackexchange.com/a/13165
+    return Math.atan2(-this[1], this[0]);
+  };
+
+  /**
+   * Applies a 2D transformation matrix (using homogeneous coordinates, so 3x3)
+   * to a Vector2 (<x, y, 1>) and returns a new vector2.
+   * @method transform
+   * @for p5.Vector
+   * @static
+   * @param {p5.Vector} v
+   * @param {p5.Transform2D} t
+   * @return {p5.Vector} a new vector
+   */
+  p5.Vector.transform = function(v, t) {
```

Comment:
Last but not least, monkeypatch `Vector.transform` and `Vector.prototype.transform` onto the p5 Vector class.  I've got these here instead of at the top with the other p5 Vector methods, because they're tightly coupled to my Transform2D class.